s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/utils/AWSV2SourceClient.java

```diff
@@ -79,46 +83,52 @@ public AWSV2SourceClient(final S3SourceConfig s3SourceConfig, final Set<String>
     }
 
     public Iterator<String> getListOfObjectKeys(final String startToken) {
-        final ListObjectsV2Request request = new ListObjectsV2Request().withBucketName(bucketName)
-                .withMaxKeys(s3SourceConfig.getS3ConfigFragment().getFetchPageSize() * PAGE_SIZE_FACTOR);
-
-        if (StringUtils.isNotBlank(startToken)) {
-            request.withStartAfter(startToken);
-        }
-        // Prefix is optional so only use if supplied
-        if (StringUtils.isNotBlank(s3SourceConfig.getAwsS3Prefix())) {
-            request.withPrefix(s3SourceConfig.getAwsS3Prefix());
-        }
+        final ListObjectsV2Request request = ListObjectsV2Request.builder()
+                .bucket(bucketName)
+                .maxKeys(s3SourceConfig.getS3ConfigFragment().getFetchPageSize() * PAGE_SIZE_FACTOR)
+                .prefix(optionalKey(s3SourceConfig.getAwsS3Prefix()))
```

Comment:
I checked, this looks like the correct way to set/unset a configuration (null/not-null)